llama/generation.py

```diff
@@ -58,33 +58,57 @@ def generate(
 
         prompt_tokens = [self.tokenizer.encode(x, bos=True, eos=False) for x in prompts]
 
-        total_len = params.max_seq_len
+        min_prompt_size = min([len(t) for t in prompt_tokens])
+        max_prompt_size = max([len(t) for t in prompt_tokens])
+        assert min_prompt_size >= 1 and max_prompt_size < params.max_seq_len
```

Comment:
shouldn't this be `max_gen_len + max_prompt_size < params.max_seq_len`?

Reply:
`max_gen_len + max_prompt_size` can be larger than `max_seq_len`, and in that case, the actual generation length would be truncated to fit in `max_seq_len`.
`max_prompt_size < params.max_seq_len` is there to make sure there is at least 1 token that can be generated for the longest prompt.